### PR TITLE
chore: Contribution guidelines and pull request template update

### PR DESCRIPTION
**Pull request by lukelooker**
Adds contribution guidelines and updates the project’s pull request template.

It should help new contributors be more successful in getting quick and effective reviews as well as help us remember all of the items that go into a successful bug fix or enhancement.

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main/CONTRIBUTING.md#developer-checklist)

- [x] 📚 Documentation updated

### CODE REVIEW

**Review by jkaster — APPROVED** [review]
Great first draft

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+```
+<type>[optional scope]: <description>
+```
+
+#### Types
+
+We generally follow [Angular's types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type). The most common types used are:
+
+- fix: a commit of the type fix patches a bug (this correlates with PATCH in Semantic Versioning).
+- feat: a commit of the type feat introduces a new feature or changes and existing one (this correlates with MINOR in Semantic Versioning).
```
Comment (bryans99) [inline CONTRIBUTING.md:53]: `change and existing`
should be
`change an existing`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+If a change does _NOT_ make a change to the build artifacts produced (`fix` or `feat` above) you can also use on of these alternative types:
```
Comment (bryans99) [inline CONTRIBUTING.md:55]: `use on of`
should be 
`use one of`

**Review by bryans99 — COMMENTED** [review]
Extremely minor typos!

**Comment by lukelooker** [comment]
@bryans99 thanks for the typo catches - added them into a follow-up PR to fix a few links #2085

**Pull request merged**